src/Drupal/ServiceMap.php

```diff
@@ -7,51 +7,51 @@
 class ServiceMap
 {
     /** @var \mglaman\PHPStanDrupal\Drupal\DrupalServiceDefinition[] */
-    private $services = [];
+    public static $services = [];
 
     public function getService(string $id): ?DrupalServiceDefinition
     {
-        // @see notes in DrupalAutoloader.
-        // This is all a work around due to inability to set container parameters.
-        if (count($this->services) === 0) {
-            $this->services = $GLOBALS['drupalServiceMap'] ?? [];
-            if (count($this->services) === 0) {
-                throw new ShouldNotHappenException('No Drupal service map was registered.');
-            }
-        }
-        return $this->services[$id] ?? null;
+        return self::$services[$id] ?? null;
     }
 
     /**
      * @return \mglaman\PHPStanDrupal\Drupal\DrupalServiceDefinition[]
      */
     public function getServices(): array
     {
-        return $this->services;
+        return self::$services;
     }
 
     public function setDrupalServices(array $drupalServices): void
     {
-        $this->services = [];
+        self::$services = [];
 
         foreach ($drupalServices as $serviceId => $serviceDefinition) {
             // @todo support factories
             if (!isset($serviceDefinition['class'])) {
                 if (isset($serviceDefinition['alias'], $drupalServices[$serviceDefinition['alias']])) {
-                    $serviceDefinition['class'] = $drupalServices[$serviceDefinition['alias']]['class'];
+                    $aliasedService = $drupalServices[$serviceDefinition['alias']];
+
+                    if (isset($aliasedService['class'])) {
+                        $serviceDefinition['class'] = $drupalServices[$serviceDefinition['alias']]['class'];
+                    } elseif (class_exists($serviceDefinition['alias'])) {
+                        $serviceDefinition['class'] = $serviceDefinition['alias'];
+                    }
+                } elseif (class_exists($serviceId)) {
+                    $serviceDefinition['class'] = $serviceId;
```

Comment:
These changes would fix https://github.com/mglaman/phpstan-drupal/issues/222, it should be spun out into a new fix if this PR cannot land easily.